anchor/anchorRecord.go

```diff
@@ -96,24 +113,49 @@ func (ar *AnchorRecord) MarshalAndSignV2(priv interfaces.Signer) ([]byte, []byte
 	return data, sig.Bytes(), nil
 }
 
-func (ar *AnchorRecord) Unmarshal(data []byte) error {
+// Non-exported, refactored function - is it better to use *string return for nil, or ""??
```

Comment:
I kept the function as it is, and changed the comment

Reply:
I didn't design v1 or v2 Anchor Records, simply added more fields to v2 to support Ethereum as well. 

But the explanation is that in V1 (for some reason), they chose to not utilize external IDs for the signature. So the content field for a ton of our bitcoin anchor records looks like this:
```
{"AnchorRecordVer":1,"DBHeight":70405,"KeyMR":"2c6c6247ae2602a08d5b59850682a60228bbc23ae66c67916836b684fc8daea9","RecordHeight":70405,"Bitcoin":{"Address":"1K2SXgApmo9uZoyahvsbSanpVWbzZWVVMF","TXID":"83ea4629b0e5d7ae7ee313d377dc4f9cb258652c76937afce4d0be29add6d8a3","BlockHeight":446482,"BlockHash":"00000000000000000166d612d5595e2b1cd88d71d695fc580af64d8da8658c23","Offset":1309}}98bb464b7b0cfd2cd43091e519193d62fc60909298761e512d34c2517f80f70eb07d5aa06ced19a85b0507cf09836141b328eb88e046bda5de021b14281a910e
```
https://explorer.factom.com/chains/df3ade9eec4b08d5379cc64270c30ea7315d8a8a1a69efe2b98a60ecdd69e604/entries/e47fe17ea16474444d3895d6048b2ade4c71114f9742d31a6e1d7d035019e2ee

They moved away from this in v2 and started using the ext-ids and content fields more sanely. But alas, immutability forces us to support old code that might not be as pretty as we'd have hoped.

Maybe a comment detailing how this is for Anchor Record V1 support, and should not be used otherwise